addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -78,29 +62,93 @@ export class DeletePlugin extends Plugin {
     // commands
     // --------------------------------------------------------------------------
 
-    deleteRange() {
-        let selection = this.shared.getEditableSelection();
+    deleteSelection() {
+        const selection = this.shared.getEditableSelection();
+        // @todo @phoenix: handle non-collapsed selection around a ZWS
+        // see collapseIfZWS
         if (selection.isCollapsed) {
             return;
         }
 
-        selection = this.deleteRangeAjustSelection(selection);
-        this.deleteSelection(selection);
+        const range = this.adjustRangeForDeletion(selection);
+
+        const { cursorPos } = this.deleteRange(range);
+
+        this.shared.setSelection(cursorPos);
     }
 
-    /**
-     *
-     * @param {import("./selection_plugin").EditorSelection} selection
-     */
-    deleteSelection(selection) {
+    deleteBackward() {
+        let selection = this.shared.getEditableSelection();
+        // Normalize selection
+        selection = this.shared.setSelection(selection);
+
+        if (!selection.isCollapsed) {
+            return this.deleteSelection();
+        }
+
+        const { endContainer, endOffset } = selection;
+        const [startContainer, startOffset] = this.findPreviousPosition(endContainer, endOffset);
+        const range = { startContainer, startOffset, endContainer, endOffset };
+
+        for (const { callback } of this.resources["handle_delete_backward"]) {
+            if (callback(range)) {
+                this.dispatch("ADD_STEP");
+                return;
+            }
+        }
+        if (!startContainer) {
+            return;
+        }
+        const { cursorPos, joined, nRemovedNodes, commonAncestor } = this.deleteRange(range);
+
+        if (nRemovedNodes || joined) {
+            this.shared.setSelection(cursorPos);
+        }
+
+        this.cleanTrailingBRs(commonAncestor);
+
+        this.dispatch("ADD_STEP");
+    }
+
+    deleteForward() {
+        let selection = this.shared.getEditableSelection();
+        // Normalize selection
+        selection = this.shared.setSelection(selection);
+
+        if (!selection.isCollapsed) {
+            return this.deleteSelection();
+        }
+
+        const { startContainer, startOffset } = selection;
+        const [endContainer, endOffset] = this.findNextPosition(startContainer, startOffset);
+        if (!endContainer) {
+            return;
+        }
+        const range = { startContainer, startOffset, endContainer, endOffset };
```

Comment:
note that it would be nicer to always manipulate a `selection` object from the selection plugin. maybe we should create a class `PhoenixSelection` that can be used easily?